common.mk

```diff
@@ -7,42 +10,69 @@ endif
 build = $(common_dir)/build.sh
 test = $(common_dir)/test.sh
 tag = $(common_dir)/tag.sh
+clean = $(common_dir)/clean.sh
 
 ifeq ($(TARGET),rhel7)
+	SKIP_SQUASH ?= 0
 	OS := rhel7
+	DOCKERFILE ?= Dockerfile.rhel7
 else ifeq ($(TARGET),fedora)
 	OS := fedora
+	DOCKERFILE ?= Dockerfile.fedora
 else
 	OS := centos7
+	DOCKERFILE ?= Dockerfile
 endif
 
+SKIP_SQUASH ?= 1
+
 script_env = \
 	SKIP_SQUASH=$(SKIP_SQUASH)                      \
 	UPDATE_BASE=$(UPDATE_BASE)                      \
 	OS=$(OS)                                        \
 	CLEAN_AFTER=$(CLEAN_AFTER)                      \
 	OPENSHIFT_NAMESPACES="$(OPENSHIFT_NAMESPACES)"
 
-.PHONY: build
-build: $(VERSIONS)
-	VERSIONS="$(VERSIONS)" $(script_env) $(tag)
+# TODO: switch to 'build: build-all' once parallel builds are relatively safe
+.PHONY: build build-serial build-all
+build: build-serial
+build-serial:
+	@$(MAKE) -j1 build-all
+
+build-all: $(VERSIONS)
+	@for i in $(VERSIONS); do \
+	    echo -n "$(BASE_IMAGE_NAME) $$i => " ; \
+	    cat $$i/.image-id ; \
+	done
 
 .PHONY: $(VERSIONS)
 $(VERSIONS): % : %/root/help.1
 	VERSION="$@" $(script_env) $(build)
 
-.PHONY: test
+.PHONY: test check
+check: test
+
 test: script_env += TEST_MODE=true
-test: $(VERSIONS)
+
+# The tests should ideally depend on $IMAGE_ID only, but see PR#19 for more info
+# while we need to depend on 'tag' instead of 'build'.
+test: tag
 	VERSIONS="$(VERSIONS)" $(script_env) $(test)
-	VERSIONS="$(VERSIONS)" $(script_env) $(tag)
 
 .PHONY: test-openshift
 test-openshift: script_env += TEST_OPENSHIFT_MODE=true
 test-openshift: $(VERSIONS)
```

Comment:
imo `test-openshift` should depend on `tag` as well

Reply:
right, good catch